### PR TITLE
Add integration tests for MacCatalyst

### LINKED ISSUE #466: Add MacCatalyst integration tests
E2E tests in the PR build like we do for the other types of apps.
Possibly run the app using `test` and `run` command.

### PR DESCRIPTION
**Pull request by premun**
Resolves #466

### CODE REVIEW

File: tests/integration-tests/Apple/Helix.SDK.Tests.proj [hunk 1]
```diff
@@ -9,15 +9,27 @@
     <HelixType>test/product/</HelixType>
     <IncludeXHarnessCli>true</IncludeXHarnessCli>
     <TestRunNamePrefix>$(AGENT_JOBNAME)</TestRunNamePrefix>
-    <EnableXUnitReporter>true</EnableXUnitReporter>
-    <EnableAzurePipelinesReporter>true</EnableAzurePipelinesReporter>
+    <EnableXUnitReporter Condition=" '$(AGENT_JOBNAME)' != '' ">true</EnableXUnitReporter>
```
Comment (MattGal) [inline tests/integration-tests/Apple/Helix.SDK.Tests.proj:12]: EnableXUnitReporter should just stay false, it doesn't actually get processed any more and can flake out and time people's tests out a tiny % of the time.

Comment (premun) [reply tests/integration-tests/Apple/Helix.SDK.Tests.proj]: Good to know! Removed from everywhere

File: tests/integration-tests/Apple/MacCatalyst.RunAppBundle.proj [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (MattGal) [inline tests/integration-tests/Apple/MacCatalyst.RunAppBundle.proj:1]: seems like these three test projects could be consolidated, not blocking your PR though.

**Review by MattGal — APPROVED** [review]

File: tests/integration-tests/Apple/MacCatalyst.RunAppBundle.proj (hunk 2 above)
Comment (premun) [reply tests/integration-tests/Apple/MacCatalyst.RunAppBundle.proj]: @MattGal good point. I found out about this `<AdditionalProperties>` property and used it to create a template `.proj` which I can now call repeatedly. Works like a charm, I even added a 4th work item!

**Pull request merged**